File: gm-utils/js/src/commom/utils/date/AnoMes.js
import IntegerFormat from '../integer/IntegerFormat';

export default class AnoMes {

	ano = 0;
	mes = 0;

	constructor(ano, mes) {
		this.ano = ano;
		this.mes = mes;
	}

	back() {
		this.mes--;
		if (this.mes === 0) {
			this.mes = 12;
			this.ano--;
		}
	}

	next() {
		this.mes++;
		if (this.mes === 13) {
			this.mes = 1;
			this.ano++;
		}
	}

	toString() {
		return IntegerFormat.xx(this.mes) + "/" + this.ano;
	}

	toInt() {
		return this.ano * 100 + this.mes;
	}

}

File: gm-utils/src/main/java/src/commom/utils/date/AnoMes.java
package src.commom.utils.date;

import src.commom.utils.integer.IntegerFormat;

public class AnoMes {

	private int ano;
	private int mes;

	public AnoMes(int ano, int mes) {
		this.ano = ano;
		this.mes = mes;
	}
	
	public void back() {
		mes--;
		if (mes == 0) {
			mes = 12;
			ano--;
		}
	}
	
	public void next() {
		mes++;
		if (mes == 13) {
			mes = 1;
			ano++;
		}
	}
	
	@Override
	public String toString() {
		return IntegerFormat.xx(mes) + "/" + ano;
	}
	
	public int toInt() {
		return ano * 100 + mes;
	}
	
}